### PR TITLE
RS/YJ/Rule 11-9

### PR DESCRIPTION
**Pull request by yunjoonjung-PNNL**
I haven't tested this rule yet.

### CODE REVIEW

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+from rct229.utils.utility_functions import find_exactly_one_service_water_heating_use
+
+
+class Section11Rule9(RuleDefinitionListIndexedBase):
+    """Rule 9 of ASHRAE 90.1-2019 Appendix G Section 11 (Service Water Heating)"""
+
+    def __init__(self):
+        super(Section11Rule9, self).__init__(
+            rmds_used=produce_ruleset_model_description(
+                USER=False, BASELINE_0=True, PROPOSED=True
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule9.py:18]: i thought this rule is only applicable to baseline? Should set USER and PROPOSED to False.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            standard_section="Table G3.1 #11, baseline column, a",
+            is_primary_rule=False,
+            list_path="ruleset_model_descriptions[0]",
+        )
+
+    class RMDRule(PartialRuleDefinition):
+        def __init__(self):
+            super(Section11Rule9.RMDRule, self).__init__(
+                rmds_used=produce_ruleset_model_description(
+                    USER=True,
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule9.py:34]: Set user to False.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                    BASELINE_0=True,
+                    PROPOSED=False,
+                ),
+            )
+
+        def applicability_check(self, context, calc_vals, data):
+            rmd_u = context.USER
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule9.py:41]: We do not need user. This is checking baseline.
The RDS seems to be a typo.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                swh_bat_u
+                for swh_bat_u in shw_bat_uses_dict_u
+                for swh_use_id in shw_bat_uses_dict_u[swh_bat_u]
+                if find_exactly_one_service_water_heating_use(rmd_b, swh_use_id).get(
+                    "use", 0.0
+                )
+                == 0.0
+            )
+
+            return f"Check that the baseline Service Water Heating System for Building Area Type {swh_bat} is sized according to ASHRAE Standard 90.1 2019, Section 7.4.1."
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule9.py:71]: I think this rule needs to rewrite to restructure the context based on the swh_bat type.
The evaluation is taking place at each `swh_bat`

You don't have to repeat the logic in `get_manual_check_required_msg` function again. The `applicability_check` true will ends in calling this function.

**Review by weilixu — CHANGES_REQUESTED** [review]
The rule definition needs to be restructured.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py (hunk 3 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule9.py:41]: Addressed.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py (hunk 2 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule9.py:34]: Addressed.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py (hunk 1 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule9.py:18]: Addressed.

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py (hunk 4 above)
Comment (yunjoonjung-PNNL) [reply rct229/rulesets/ashrae9012019/section11/section11rule9.py:71]: I've addressed the 1st part.

I'm a bit confused the second part ```You don't have to repeat the logic in get_manual_check_required_msg function again. The applicability_check true will ends in calling this function.``` - Could you be a bit more specific?

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                each_rule=Section11Rule9.RMDRule.SWHBATRule(),
+            )
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+
+            service_water_heating_uses_dict_b = {
+                swh_use["id"]: swh_use.get("use", 0.0)
+                for swh_use in find_all(
+                    "$.buildings[*].building_segments[*].zones[*].spaces[*].service_water_heating_uses[*]",
```
Comment (weilixu) [inline rct229/rulesets/ashrae9012019/section11/section11rule9.py:47]: Add a comment about the schema change.

**Review by weilixu — APPROVED** [review]

File: rct229/rulesets/ashrae9012019/section11/section11rule9.py (hunk 4 above)
Comment (weilixu) [reply rct229/rulesets/ashrae9012019/section11/section11rule9.py]: It's all good. I approved the PR. Thank you Yun Joon

**Pull request merged**